float8_experimental/float8_linear.py

```diff
@@ -450,14 +460,18 @@ def from_float(
         scaling_type_input=TensorScalingType.DYNAMIC,
         scaling_type_weight=TensorScalingType.DYNAMIC,
         scaling_type_grad_output=TensorScalingType.DYNAMIC,
+        config: Optional[Float8LinearConfig] = None,
     ):
         """
         Create an nn.Linear with fp8 compute from a regular nn.Linear
 
         Args:
             mod (torch.nn.Linear): nn.Linear to convert
             emulate (bool): whether to emulate fp8 matmul logic in float32
+            config (Optional[Float8LinearConfig]): configuration for conversion to float8
         """
+        if config is None:
+            config = Float8LinearConfig()
```

Comment:
is non-None config the key to avoid torch.compile error?

Reply:
I'm just taking a shortcut:
1. long term config should be required
2. short term, making config required would need us to move it up to positional args, which would then require us to modify all the callsites
3. long term, we will also move `emulate` and `scaling_type...` to `config`

If we make the argument required after we do 3, it should be less work overall.